test/src/unittests/tonal/test_pitchsaliencefunctionpeaks.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python
+
+# Copyright (C) 2006-2021  Music Technology Group - Universitat Pompeu Fabra
+#
+# This file is part of Essentia
+#
+# Essentia is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Affero General Public License as published by the Free
+# Software Foundation (FSF), either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the Affero GNU General Public License
+# version 3 along with this program. If not, see http://www.gnu.org/licenses/
+
+
+from numpy import *
+from essentia_test import *
+
+class TestPitchSalienceFunctionPeaks(TestCase):
+  
+    def testInvalidParam(self):
+        self.assertConfigureFails(PitchSalienceFunctionPeaks(), {'binResolution': -1})
+        self.assertConfigureFails(PitchSalienceFunctionPeaks(), {'maxFrequency': -1})
+        self.assertConfigureFails(PitchSalienceFunctionPeaks(), {'minFrequency': -1})
+        self.assertConfigureFails(PitchSalienceFunctionPeaks(), {'referenceFrequency': -1})                
+
+    def testEmpty(self): 
+        self.assertRaises(RuntimeError, lambda: PitchSalienceFunctionPeaks()([]))
+    
```

Comment:
Here we are missing a test to validate that the input vector size corresponds to the `binResolution`, `referenceFrequency`, and `maxFrequency`. What would happen if the input size is [smaller that expected](https://github.com/MTG/essentia/blob/master/src/algorithms/tonal/pitchsaliencefunctionpeaks.cpp#L47) according to these parameters?

Reply:
If binResolution is set to a high value and input size is small, a "nan" is returned in zeroth bin element.
    
    See code extract below
    
    """ FIXME

    The following test case has uncovered a situation where  divide by zero or some other 
    illegal operation has taken place because a NAN vlaue is found in output array element zero.

    def testBinResolutionTooHigh(self):        
        # Choose a sample set of frequencies and magnitude vectors of unequal length
        freq_speaks = [55] 
        mag_speaks = [1] 
        calculatedPitchSalience = PitchSalienceFunction(binResolution=55*2)(freq_speaks,mag_speaks)       
        print(calculatedPitchSalience)

        result:  calculatedPitchSalience[0] contains "nan"

    """